tests/test_modelrun.py

```diff
@@ -1,132 +1,31 @@
-                    'value': 3,
-                    'region': 'oxford',
-                    'interval': 1
-                },
-                {
-                    'year': 2011,
-                    'value': 5,
-                    'region': 'oxford',
-                    'interval': 1
-                },
-                {
-                    'year': 2012,
-                    'value': 1,
-                    'region': 'oxford',
-                    'interval': 1
-                }
-            ]
-        },
-        "region_sets": {'BSOA': setup_region_data['features']},
-        "interval_sets": {
-            'yearly': [
-                {
-                    'id': 1,
-                    'start': 'P0Y',
-                    'end': 'P1Y'
-                }
-            ]
-        },
-        "scenario_metadata": [
-            {
-                'name': 'raininess',
-                'temporal_resolution': 'yearly',
-                'spatial_resolution': 'BSOA',
-                'units': 'ml'
-            }
-        ]
-    }
-
+def get_model_run_config_data():
 
-@fixture(scope='function')
-def get_model_run(setup_project_folder, setup_region_data):
-    path = setup_project_folder
-    water_supply_wrapper_path = str(
-        path.join(
-            'models', 'water_supply', '__init__.py'
-        )
-    )
-
-    config_data = {
+    config = {
+        'name': 'unique_model_run_name',
+        'stamp': '2017-09-20T12:53:23+00:00',
+        'description': 'a description of what the model run contains',
         'timesteps': [2010, 2011, 2012],
-        'dependencies': [],
-        'region_sets': {},
-        'interval_sets': {},
-        'planning': [],
-        'scenario_metadata':
-            [{
-                'name': 'raininess',
-                'temporal_resolution': 'yearly',
-                'spatial_resolution': 'BSOA',
-                'units': 'ml'
-            }],
-        'scenario_data': {
-            "raininess": [
-                {
-                    'year': 2010,
-                    'value': 3,
-                    'region': 'oxford',
-                    'interval': 1
-                },
-                {
-                    'year': 2011,
-                    'value': 5,
-                    'region': 'oxford',
-                    'interval': 1
-                },
-                {
-                    'year': 2012,
-                    'value': 1,
-                    'region': 'oxford',
-                    'interval': 1
-                }
-            ]
-        },
-        "sector_model_data": [
-            {
-                "name": "water_supply",
-                "path": water_supply_wrapper_path,
-                "classname": "WaterSupplySectorModel",
-                "inputs": [],
-                "outputs": [],
-                "initial_conditions": [],
-                "interventions": [],
-                "parameters": []
-            }
-        ]
+        'sos_model': Mock(sector_models=[]),
```

Comment:
So a builder expects a single `SoSModel` in the config and it is the controller's job to read the right model for the `ModelRun`, switching out reference-by-id for object before passing the config to the builder. Looks sound.

Reply:
Yep